presto-kafka/src/main/java/io/prestosql/plugin/kafka/schema/confluent/SchemaParser.java

```diff
@@ -13,10 +13,11 @@
  */
 package io.prestosql.plugin.kafka.schema.confluent;
 
+import io.confluent.kafka.schemaregistry.ParsedSchema;
 import io.prestosql.plugin.kafka.KafkaTopicFieldGroup;
 import io.prestosql.spi.connector.ConnectorSession;
 
 public interface SchemaParser
 {
-    KafkaTopicFieldGroup parse(ConnectorSession session, String subject, String rawSchema);
+    KafkaTopicFieldGroup parse(ConnectorSession session, String subject, ParsedSchema parsedSchema);
```

Comment:
Make `ParsedSchema` a generic. That way the type check and cast in `AvroSchemaParser` is not needed

Reply:
But `ParsedSchema` is from `SchemaRegistry` and it is given by its client